lib/bundler/definition.rb

```diff
@@ -907,10 +909,16 @@ def expand_dependencies(dependencies, remote = false)
       deps
     end
 
+    def dependencies_for(groups)
+      current_dependencies.reject do |d|
+        (d.groups & groups).empty?
+      end
+    end
```

Comment:
It's a helper method to use the common code in this class to choose valid dependencies.